play-route/src/main/scala/org/http4s/server/play/PlayRouteBuilder.scala

```diff
@@ -0,0 +1,149 @@
+package org.http4s.server.play
+
+import akka.stream.scaladsl.{Sink, Source}
+import akka.util.ByteString
+import cats.data.OptionT
+import cats.effect.{Effect, IO}
+import fs2.interop.reactivestreams._
+import org.http4s.server.play.PlayRouteBuilder.{PlayAccumulator, PlayRouting, PlayTargetStream}
+import org.http4s.{EmptyBody, Header, Headers, HttpService, Method, Request, Response, Uri}
+import play.api.http.HttpEntity.Streamed
+import play.api.libs.streams.Accumulator
+import play.api.mvc._
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future, Promise}
+
+class PlayRouteBuilder[F[_]](
+    service: HttpService[F]
+)(implicit F: Effect[F], executionContext: ExecutionContext) {
+
+  type UnwrappedKleisli = Request[F] => OptionT[F, Response[F]]
+  private[this] val unwrappedRun: UnwrappedKleisli = service.run
+
+  def requestHeaderToRequest(requestHeader: RequestHeader): Request[F] =
+    Request(
+      // todo fix the ???
+      method = Method.fromString(requestHeader.method).getOrElse(???),
```

Comment:
You can make this whole method return a `ParseResult[Request[F]]` and deal with errors closer to the seem with Play.  That would be the more FP way.  Or you might just throw an exception here, which (I think) is more the Play way.  Since you're fusing idioms, there's no obvious right answer to me.

Reply:
Yep I'll have to flatten out the error somewhere.